src/third-party/PhishingController.test.ts

```diff
@@ -1,350 +1,597 @@
+      .reply(200, []);
+
+    const controller = new PhishingController();
     await controller.test('metamask.io');
-    expect(mock.callCount).toBe(1);
-    await clock.tickAsync(2);
+    await clock.tickAsync(defaultRefreshInterval);
     await controller.test('metamask.io');
-    expect(mock.callCount).toBe(2);
-    clock.restore();
-    mock.restore();
+
+    expect(nockScope.isDone()).toBe(true);
   });
 
   it('should be able to change the refreshInterval', async () => {
-    const mock = sinon.spy(PhishingController.prototype, 'updatePhishingLists');
-    const controller = new PhishingController({});
-    controller.setRefreshInterval(0);
+    const nockScope = nock('https://cdn.jsdelivr.net')
+      .get('/gh/MetaMask/eth-phishing-detect@master/src/config.json')
+      .times(3)
+      .reply(200, {
+        blacklist: [],
+        fuzzylist: [],
+        tolerance: 0,
+        whitelist: [],
+        version: 0,
+      })
+      .get('/gh/phishfort/phishfort-lists@master/blacklists/hotlist.json')
+      .times(3)
+      .reply(200, []);
 
+    const controller = new PhishingController({ refreshInterval: 1 });
+    controller.setRefreshInterval(0);
     await controller.test('metamask.io');
     await controller.test('metamask.io');
     await controller.test('metamask.io');
 
-    expect(mock.callCount).toBe(3);
-    mock.restore();
+    expect(nockScope.isDone()).toBe(true);
   });
 
   it('should update lists', async () => {
+    const mockPhishingBlocklist = ['https://example-blocked-website.com'];
+    const phishfortHotlist = ['https://another-example-blocked-website.com'];
+    nock('https://cdn.jsdelivr.net')
+      .get('/gh/MetaMask/eth-phishing-detect@master/src/config.json')
+      .reply(200, {
+        blacklist: mockPhishingBlocklist,
+        fuzzylist: [],
+        tolerance: 0,
+        whitelist: [],
+        version: 0,
+      })
+      .get('/gh/phishfort/phishfort-lists@master/blacklists/hotlist.json')
+      .reply(200, phishfortHotlist);
+
     const controller = new PhishingController();
-    controller.update({}, true);
     await controller.updatePhishingLists();
-    controller.state.phishing.forEach((config) => {
-      expect(config).toHaveProperty('allowlist');
-      expect(config).toHaveProperty('blocklist');
-      expect(config).toHaveProperty('fuzzylist');
-      expect(config).toHaveProperty('tolerance');
-      expect(config).toHaveProperty('name');
-      expect(config).toHaveProperty('version');
-    });
+
+    expect(controller.state.phishing).toStrictEqual([
+      {
+        allowlist: [],
+        blocklist: mockPhishingBlocklist,
+        fuzzylist: [],
+        tolerance: 0,
+        name: 'MetaMask',
+        version: 0,
+      },
+      {
+        allowlist: [],
+        blocklist: phishfortHotlist,
+        fuzzylist: [],
+        tolerance: 0,
+        name: 'PhishFort',
+        version: 1,
+      },
+    ]);
   });
 
   it('should not update infura rate if disabled', async () => {
```

Comment:
Unsure what this test is supposed to test.

Reply:
It's supposed to test that the lists do not get updated when the controller is disabled. I too am confused by what "infura rate" is meant to mean here. I can reword it.

Reply:
Glad it wasn't just me 😅